packages/views/src/components/theme-provider.tsx

```diff
@@ -1,81 +1,66 @@
-import { createContext, useContext, useEffect, useState } from 'react'
+import { useEffect } from 'react'
 
-type Theme = 'dark' | 'light' | 'system'
+import { create } from 'zustand'
+import { persist } from 'zustand/middleware'
 
-type ThemeProviderProps = {
-  children: React.ReactNode
-  defaultTheme?: Theme
-  storageKey?: string
-}
-
-type ThemeProviderState = {
-  theme: Theme
-  setTheme: (theme: Theme) => void
-}
+export type ModeType = 'dark' | 'light' | 'system'
+export type ColorType = 'std' | 'tri' | 'pnd'
+export type ContrastType = 'standard' | 'low' | 'high'
+export type FullTheme = `${ModeType}-${ColorType}-${ContrastType}`
 
-const initialState: ThemeProviderState = {
-  theme: 'system',
-  setTheme: () => null
+type ThemeState = {
+  theme: FullTheme
+  setTheme: (theme: FullTheme) => void
 }
 
-const ThemeProviderContext = createContext<ThemeProviderState>(initialState)
-
-export function ThemeProvider({
-  children,
-  defaultTheme = 'system',
-  storageKey = 'canary-ui-theme',
-  ...props
-}: ThemeProviderProps) {
-  const getMediaQuery = () => window.matchMedia('(prefers-color-scheme: dark)')
+const useThemeStore = create<ThemeState>()(
+  persist(
+    set => ({
+      theme: 'system-std-standard', // Default theme
+      setTheme: (newTheme: FullTheme) => set({ theme: newTheme })
+    }),
+    {
+      name: 'canary-ui-theme' // LocalStorage key
+    }
+  )
+)
 
-  const [theme, setThemeState] = useState<Theme>(() => (localStorage.getItem(storageKey) as Theme) || defaultTheme)
-  const [systemTheme, setSystemTheme] = useState<'dark' | 'light'>(() => (getMediaQuery().matches ? 'dark' : 'light'))
+export function ThemeProvider({ children }: { children: React.ReactNode }) {
+  const theme = useThemeStore(state => state.theme)
+  const setTheme = useThemeStore(state => state.setTheme)
```

Comment:
is there a way to destructure both in the same line? 
```
const { theme, setTheme } = useThemeStore(state => ({
  theme: state.theme,
  setTheme: state.setTheme,
}))

```

Reply:
Makes sense, done!